crates/net/peers/src/node_record.rs

```diff
@@ -87,9 +87,12 @@ impl NodeRecord {
         self
     }
 
-    /// Creates a new record from a socket addr and peer id.
-    pub const fn new(addr: SocketAddr, id: PeerId) -> Self {
-        Self { address: addr.ip(), tcp_port: addr.port(), udp_port: addr.port(), id }
+    /// Creates a new record from a tcp and udp socket and peer id.
+    pub const fn new(tcp_addr: SocketAddr, udp_addr: Option<SocketAddr>, id: PeerId) -> Self {
```

Comment:
this change is very invasive, I'd like to keep the previous new function.

the upd/tcp port can be set separately via `f with_*`

Reply:
yeah, but that seems a bit lengthy, `NodeRecord::new(xxx).with_udp_port(xxx)`